### PR TITLE
MAINT: Updating SciPy gitpod

### PR DESCRIPTION
**Pull request by trallard**
#### Reference issue

Related to #13677

#### What does this implement/fix?

This PR tackles the issues mentioned in the above issue and makes improvements to the GitPod integration

**Editor**
- [x] Make sure the Python extension is installed -> since we are using our custom image
- [x] Remove deprecated settings in VSCode (`PythonPath`)

**Docker images**
- [x]  Add the gh cli (https://github.com/cli/cli) to the gitpod container
- [x] Build Gitpod image nightly instead of on pushes to master (in sync with SciPy builds)
- [x] Add VSCode extension `RestructuredText` and `sphinx-autobuild` -> should remove the rst warning when initialising a workspace
- [x] Added use of ccache (thanks to @thomasjpfan for the inspiration) to speed up build in Gitpod

**GitHub actions**
- [x] Cache image layers to reduce build times
- [x] Base image now only gets built and pushed when there are changes to `environment.yml` -> Deleted `environment.yml` to root so that it can be overwritten by @rgommers update later on
- [x] Add workflow to build and push the gitpod Docker image (`.github/workflows/docker.yml`) -> this is the built nightly

**Gitpod**
- [x] Use `scipy/scipy-gitpod:latest` which is built nightly
- [x] ensuring that the most recent prebuild is picked -> building SciPy in Docker container then rebuilding in init so no need to rely on prebuild as a workaround now
- [x]  re-run of `make html-scipyorg` now works as expected

#### Additional information

- [ ]  see if we can name scipy/scipy as upstream rather than origin, to match our contributor guide -> no solution atm though removing the dependency on the build enables folks to work directly from their forks :warning: They seemed to have changed things again on the Gitpod side so this does not seem doable atm; workaround would be to have folks fork and work from those forks
- Sending separate PR for documentation on how to use GitPod 📖

### CODE REVIEW

**Comment by trallard** [comment]
Snapshot of my current workspace: https://gitpod.io#snapshot/d669f48a-02ec-43bf-bbc7-9460257c248d

**Comment by rgommers** [comment]
Looks promising, thanks @trallard! This has a merge conflict on `environment.yaml`. I think you can simply delete it in this PR rather than move it; what's now in `environment.yml` in the root of the repo includes everything that the one in `tools/docker_dev` had.

**Comment by tupui** [comment]
If `environment.yml` is deleted (what I understand reading your comment Ralf), then don't forget about updating the documentation instructions (conda quick start).

**Review by thomasjpfan — COMMENTED** [review]
This looks good! Building the first Dockerfile with only dependencies is a nice way to speed up building the gitpod image.

File: .gitpod.yml [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
-image:
-  file: ./tools/docker_dev/.gitpod.Dockerfile
+image: scipy/scipy-gitpod:latest
 tasks:
-  - name: Prebuild Scipy & docs
-    prebuild: |
-      conda activate scipydev
+  - name: Prepare development
+    init: |
+      conda activate scipy-dev
```
[the remark this answers is not in the document]
Comment (trallard) [reply .gitpod.yml:12]: I should have removed it  - missed this!

**Comment by trallard** [comment]
If the workspace config + extensions are working fine we can carry those over to Numpy as well as making some adjustments to the Docker images

**Pull request closed without merging**